### PR TITLE
Qt regressions

### PR DESCRIPTION
**Pull request by spireken**
I used the Book View as an example and put it in the Spire Desktop drafts folder.

I'll make another branch for the window regression.

### CODE REVIEW

**Comment by spireken** [comment]
I added the Toolbar Window demo to the drafts folder as well.

File: Applications/Spire/Include/Spire/Toolbar/MenuIconSizeProxyStyle.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#ifndef SPIRE_MENU_ICON_SIZE_PROXY_STYLE_HPP
+#define SPIRE_MENU_ICON_SIZE_PROXY_STYLE_HPP
+#include <QProxyStyle>
+
+namespace Spire {
+
+  class MenuIconSizeProxyStyle : public QProxyStyle {
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Toolbar/MenuIconSizeProxyStyle.hpp]: Document.

File: Applications/Spire/Include/Spire/Toolbar/MenuIconSizeProxyStyle.hpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public:
+
+      MenuIconSizeProxyStyle(int size);
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Toolbar/MenuIconSizeProxyStyle.hpp]: Mark ```explicit```, feel free to read up on why.

File: Applications/Spire/Include/Spire/Toolbar/MenuIconSizeProxyStyle.hpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      int pixelMetric(QStyle::PixelMetric metric,
+        const QStyleOption* option = nullptr,
+        const QWidget* widget = nullptr) const override;
+
+    private:
+      int m_size;
+  };
+}
+
+#endif
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Toolbar/MenuIconSizeProxyStyle.hpp]: End files with a new line.

File: Applications/Spire/Source/Toolbar/ToolbarMenu.cpp [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #include "Spire/Toolbar/ToolbarMenu.hpp"
+#include "Spire/Toolbar/MenuIconSizeProxyStyle.hpp"
```
Comment (kranar) [inline Applications/Spire/Source/Toolbar/ToolbarMenu.cpp]: Alphabetical order except for line 1.

File: Applications/Spire/Source/Toolbar/ToolbarMenu.cpp [hunk 5]
```diff
@@ -9,8 +10,10 @@ using namespace Spire;
 ToolbarMenu::ToolbarMenu(const QString& title, QWidget* parent)
     : QPushButton(title, parent),
       m_items(new QMenu(this)),
+      m_menu_icon_style(new MenuIconSizeProxyStyle(scale_width(18))),
```
Comment (kranar) [inline Applications/Spire/Source/Toolbar/ToolbarMenu.cpp]: Use a ```unique_ptr```.

File: Applications/Spire/Source/Toolbar/ToolbarMenu.cpp [hunk 6]
```diff
@@ -124,10 +131,10 @@ void ToolbarMenu::set_empty_menu_stylesheet() {
       font-style: italic;
       height: %2px;
       padding-left: %3px;
-  })").arg(scale_height(12)).arg(scale_height(20)).arg(scale_width(8)));
+    })").arg(scale_height(12)).arg(scale_height(20)).arg(scale_width(8)));
```
Comment (kranar) [inline Applications/Spire/Source/Toolbar/ToolbarMenu.cpp]: Fix indentation.

File: Applications/Spire/Source/Ui/Ui.cpp [hunk 7]
```diff
@@ -10,10 +10,12 @@ QImage Spire::imageFromSvg(const QString& path, const QSize& size) {
...
-  auto renderer = QSvgRenderer(path);
+  // TODO: Revert this when Qt fixes the regression.
+  // https://bugreports.qt.io/browse/QTBUG-81259
+  auto svg_pixmap = QIcon(path).pixmap(box.width(), box.height());
   auto image = QImage(size, QImage::Format_ARGB32);
   image.fill(QColor(0, 0, 0, 0));
-  QPainter painter(&image);
-  renderer.render(&painter, box);
+  auto painter = QPainter(&image);
+  painter.drawPixmap(QPoint(box.x(), box.y()), svg_pixmap);
```
Comment (kranar) [inline Applications/Spire/Source/Ui/Ui.cpp]: Use ```box.topLeft()```.

**Pull request merged**